New Left Review.js

```diff
@@ -0,0 +1,257 @@
+	"label": "New Left Review",
+	"creator": "Bo An",
+	"target": "^https?://(www.)?newleftreview.org/issues/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2021-06-12 07:59:12"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+	Copyright © 2021 Bo An
+	This file is part of Zotero.
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc, url) {
+	const article = "articles";
+	const isArticle = url.includes(article);
+	if (isArticle) {
+		return 'journalArticle';
+	}
+	else {
+		return 'multiple';
+	}
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), (items) => {
+			if (!items) {
+				return true;
+			}
+			const articles = [];
+			for (const i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+			return true;
+		});
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+function getSearchResults(doc) {
+	let items = {};
+	let found = false;
+
+	const articleLinks = ZU.xpath(doc, '//a[@class="article-link"]');
+	const titles = ZU.xpath(doc, '//span[@class="programme-note__title"]');
+	const authors = ZU.xpath(doc, '//span[@class="programme-note__author"]');
+
+	articleLinks.forEach((articleLink, articleIndex) => {
+		const href = articleLink.href;
+		let title = titles[articleIndex].innerText;
+		let author = authors[articleIndex];
+		if (author) {
+			const authorNames = author.innerText;
+			title += ` (${authorNames})`;
+		}
+		items[href] = title;
+		if (found === false) {
+			found = true;
+		}
+	});
+
+	return found ? items : false;
+}
+
+function scrape(doc, _) {
+	const newItem = new Zotero.Item('journalArticle');
+
+	newItem.publicationTitle = "New Left Review";
+	newItem.journalAbbreviation = "NLR";
+
+	const title = ZU.xpath(doc, '//meta[@itemprop="name"]/@content')[0].value;
+
+	newItem.title = title;
+
+	const description = ZU.xpathText(doc, '//meta[@name="description"]/@content');
+
+	if (description) {
+		newItem.abstractNote = description;
+	}
+
+	const authors = ZU.xpath(doc, '//meta[@itemprop="author"]/@content');
+
+	const authorFullNames = authors.map(author => author.value);
+
+	authorFullNames.forEach((authorFullName) => {
+		newItem.creators.push({
```

Comment:
Is there a reason we should use `fieldMode: 1` and not use `ZU.cleanAuthor` here?

Reply:
Thanks! I wasn't aware of this utility class. Super helpful. 

I was wondering if there is a similar utility class for titles too (to convert all caps to the title case for example)?

Reply:
Yes! There's [`ZU.capitalizeTitle`](https://github.com/zotero/zotero/blob/f1414731daa749c1de7b5946e9c08b571c710d1e/chrome/content/zotero/xpcom/utilities.js#L903).

Reply:
In a lot of cases, when you *know* the titles are screwed up, you'll want to pass `true` as the second argument to that function (`force`). Otherwise it'll only perform the transformation if the user has manually enabled the `capitalizeTitles` preference.

Reply:
Thanks for the tips Abe! I'll try using it next time.